Brief-over-of-Heads-and-firmware-security/FAQ.md

```diff
@@ -0,0 +1,139 @@
+ for building slightly more secure systems. Installing Heads requires opening
+ the machine and extensive warranty voiding, so this document tries to answer
+ some of the questions about it.
+
+Why replace UEFI with coreboot
+----
+
+While Intel's edk2 tree that is the base of UEFI firmware is open source, the
+ firmware that vendors install on their machines is proprietary and closed
+ source. Updates for bugs fixes or security vulnerabilities are at the vendor's
+ convenience; user specific enhancements are likely not possible; and the code
+ is not auditable.
+
+UEFI is much more complex than the BIOS that it replaced. It consists of
+ millions of lines of code and is an entire operating system, with network
+ device drivers, graphics, USB, TCP, https, etc, etc, etc. All of these features
+ represents increased "surface area" for attacks, as well as unnecessary
+ complexity in the boot process.
+
+coreboot is open source and focuses on just the code necessary to bring the
+ system up from reset. This minimal code base has a much smaller surface area
+ and is possible to audit. Additionally, self-help is possible if custom
+ features are required or if a security vulnerability needs to be patched.
+
+What's wrong with UEFI Secure Boot
+----
+
+Can't audit it, signing keys are controlled by vendors, doesn't handle hand off
+ in all cases, depends on possible leaked keys.
+
+Why use Linux instead of vboot2
+----
+
+vboot2 is part of the coreboot tree and is used by Google in the Chromebook
+ system to provide boot time security by verifying the hashes on the coreboot
+ payload. This works well for the specialized Chrome OS on the Chromebook, but
+ is not as flexible as a measured boot solution.
+By moving the verification into the boot scripts we're able to have a much
+ flexible verification system and use more common tools like PGP to sign
+ firmware stages.
+
+What about Trusted GRUB
+----
+
+The mainline grub doesn't have support for TPM and signed kernels, but there is
+ a Trusted grub fork that does. Due to philosophical differences the code might
+ not be merged into the mainline. And due to problems with secure boot (which
+ Trusted Grub builds on), many distributions have signed insecure kernels
+ that bypass all of the protections secure boot promised.
+Additionally, grub is closer to UEFI in that it must have device drivers for
+ all the different boot devices, as well as filesystems. This duplicates the
+ code that exists in the Linux kernel and has its own attack surface.
+
+Using coreboot and Linux as a boot loader allows us to restrict the signature
+ validation to keys that we control. We also have one code base for the device
+ drivers in the Linux-as-a-boot-loader as well as Linux in the operating system.
+
+What is the concern with the Intel Management Engine
+----
+
+"Rootkit in your chipset", "x86 considered harmful", etc
+
+How about the other embedded devices in the system
+----
+
+goodbios, funtenna, etc.
```

Comment:
Same here, searched real quick but references are not in my history so I can't refer to the talk/papers here. Would add (TODO: reference missing to welcome contributions, here and everywhere else where refs are missing)

Reply:
goodbios ref https://github.com/osresearch/heads-wiki/pull/49/files/1899083cd24a471d83b67df360b4a174165d8bd3#diff-89e91d5fe0f3cf3e8289a4a683dd2108R15